src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -7,6 +7,7 @@
 // the Business Source License, use of this software will be governed
```

Comment:
I think this could be two or three at most separate commits. IMO the general gist of the commit is to refactor, and it'd be unnecessarily verbose in the commit tree to have a dozen extra tiny commits. I think this change is suffering from it not really ever being obvious to review a batch of small changes -- it's hard to see the relevance or why it's useful in the diff, but it's easier to see when looking at the before and after of a function.

This also begs the question of why, when introducing a new package, each function or struct introduced isn't a separate commit.

I think this commit is relatively easy to split into two or three, and I'll do so. One of the commits may still be trickier than the others.

Reply:
This has been split.